### PR TITLE
Podcast Block: Save RSS feed URL as post content (#15682)

### LINKED ISSUE #15682: Podcast Player: add save
We currently do not save anything to post content when you add a block to a post. This is problematic for folks who may deactivate Jetpack later on, and for folks accessing the post outside of the WordPress context (RSS feeds, subscription emails, ...)

I think we should aim to have something simple saved in the post content, such as  a link to the RSS feed at least.

### PR DESCRIPTION
**Pull request by mreishus**
Fixes #15682

#### Changes proposed in this Pull Request:

* The podcast player now saves a simple RSS link to the post content.  This has no effect on regular use of the block, but is shown if the block is no longer active for some reason (such as disabling jetpack).

#### Jetpack product discussion

N/A

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

**CLI Unit Tests**

* Run `yarn test-extensions` in jetpack root.

**Interface test**

You need a way to publish a post using the podcast block from the jetpack plugin, then a way to either disable the jetpack plugin or the podcast block.  I found it impossible to disable jetpack on my sandbox, so I ended up doing this:

* Load branch
* Run `yarn build-extensions --watch` and set up a real-time sync to sandbox
* Create a new post, add a podcast block, add a valid podcast RSS (example: `http://feeds.99percentinvisible.org/`), publish the post
* View the post.  Podcast should display properly
* Disable the podcast block.  Temporarily edit `extensions/index.json` and remove `"podcast-player"`.  Let the sandbox sync.
* View the post.  The block should be inactive and you should see the saved data (a simple link to the RSS)

#### Screenshots from my test

**Podcast block active:**
podcast-active-small

**Podcast block inactive:**
podcast-inactive

#### Proposed changelog entry for your changes:

* Podcast Player: Add save function

### CODE REVIEW

**Comment by mreishus** [comment]
I'm not sure why the pre-commit hook was skipped.  I used `git --amend` to recommit, letting the husky pre-commit hooks run, then `git push --force-with-lease` to update the branch to mark that the pre-commit hook was run.

**Comment by mreishus** [comment]
Thanks.  In my testing, I was able to recreate the migration error.

I created a simple deprecated v1 in 41c6754, which omitted the optional `migrate()` function, and the site was able to upgrade an old block properly.



I also verified that removing the podcast-player block after the upgrade would result in the site rendering the RSS link.

---

_Note_:  I had to duplicate the identical `supports`, which I wasn't 100% happy about:  [current supports](https://github.com/Automattic/jetpack/blob/41c675490a0d7d9a4ebafee167c20be4a9e92481/extensions/blocks/podcast-player/index.js#L37) | [deprecated v1 supports](https://github.com/Automattic/jetpack/blob/41c675490a0d7d9a4ebafee167c20be4a9e92481/extensions/blocks/podcast-player/deprecated/v1/index.js#L11).

I tried to import them, but I was getting some baffling behavior:
```js
// in extensions/blocks/podcast-player/deprecated/v1/index.js

import { settings, name, title } from '../../index';
console.log( '== ??? ' );
console.log( { settings, name, title } );
console.log( '==' );
```

Resulted in console output: `Object { settings: undefined, name: undefined, title: undefined }`.  [However, name, title and settings are exported from that file.](https://github.com/Automattic/jetpack/blob/41c675490a0d7d9a4ebafee167c20be4a9e92481/extensions/blocks/podcast-player/index.js#L25-L28)  No matter what I did, I couldn't seem to import from `podcast-player/index.js` inside `podcast-player/deprecated/v1/index.js`.  But importing from an adjacent file like `attributes.js` was fine. Not sure if this is a quirk of my build+rsync setup, or perhaps the deprecation system itself.

I opted to duplicate the identical supports.  Another possible approach would be to extract them into a `supports.js` file and to have both the deprecated v1 and current import from that file.

**Comment by marekhrabe** [comment]
> I opted to duplicate the identical supports. 

That's actually exactly how deprecations should work - copied code all over. They shouldn't share any code with other parts as they are rather a "snapshot" of the block at a certain time. 

On the contrary, I think you should also copy the whole `attributes` object. Otherwise, if you later add a new attribute into `attributes.js`, you have also edited the deprecation declaration and suddenly, with the additional attribute, it will no longer match what the block originally looked like and the whole deprecation will be broken.

File: extensions/blocks/podcast-player/test/save.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+};
+
+describe( 'save', () => {
+	test( 'Save generates a simple link to the RSS url', () => {
+		// Verify the output of the save block, given the example settings.
+		//
+		// This is close to a snapshot test; done because actual snapshots don't work for testing
+		// block save() functions.  See ../../../README.md and https://github.com/Automattic/wp-calypso/pull/30727
+		const want = (
+			<a className="jetpack-podcast-player__direct-link" href="http://example.com/podcast/feed.rss">
```
Comment (jeryj) [inline extensions/blocks/podcast-player/test/save.js:19]: @getdave A test! 🎉 Want to take a peek?

**Comment by mreishus** [comment]
> On the contrary, I think you should also copy the whole attributes object.

Added 43839440fc, which keeps a static copy of the `attributes` object.  Re-tested to make sure the upgrade process still works, and that viewing the post after removing the podcast-player shows the link to the rss.

**Comment by mreishus** [comment]
Some of the travis builds are failing, but the same pattern of failures are happening on the master branch as well.  So they're not unique to this branch.

Latest builds on travis as now:
This branch: https://travis-ci.com/github/Automattic/jetpack/builds/176529639
Master: https://travis-ci.com/github/Automattic/jetpack/builds/176509784

**Comment by jeherve** [comment]
> Some of the travis builds are failing

This is caused by some changes in Core. We're tracking this in #16539, hopefully will get this sorted out soon.

**Comment by mreishus** [comment]
Ok, oops! I guess the jetpack crew review is the next step in the process.

**Comment by marekhrabe** [comment]
Correct, somebody from the Jetpack Crew will give a final 👍 and then we proceed with merging.

**Comment by mreishus** [comment]
Planning to merge+deploy on Monday 2020-07-27.

**Comment by jeherve** [comment]
r211581-wpcom

**Pull request merged**